docs/node.md

```diff
@@ -0,0 +1,140 @@
+# Node
+
+The Node struct stores a node in the IAVL tree. 
+
+### Structure
+
+```golang
+// Node represents a node in a Tree.
+type Node struct {
+	key       []byte // key for the node.
+	value     []byte // value of leaf node. If inner node, value = nil
+	version   int64  // The version of the IAVL that this node was first added in.
+	height    int8   // The height of the node. Leaf nodes have height 0
+	size      int64  // The number of leaves that are under the current node. Leaf nodes have size = 1
+	hash      []byte // hash of above field and leftHash, rightHash
+	leftHash  []byte // hash of left child
+	leftNode  *Node  // pointer to left child
+    rightHash []byte // hash of right child
+	rightNode *Node  // pointer to right child
+	saved     bool   // saved to memory or disk
+	persisted bool   // persisted to disk
```

Comment:
what's the difference between saved and persisted? (feel free to point me elsewhere in this doc if it's been answered already!)